### DISCUSSION TITLE
Character range pattern matching using integer literals

### DISCUSSION THREAD

**Discussion by Rekkonnect** [dotnet/csharplang#6350] [Language Ideas]
I came across a scenario where defining extensions for code pages required using the specialized Unicode character syntax, like so
```csharp
return c is >= '\u0250' and <= '\u02FF';
```
~~There is also the possibility of using an integer over 4 hex digits (`10EAFF` for example), which would require using the 8-hex digit syntax (`\u0010EAFF`). That causes convenience problems in itself, as what's being compared in the end is the integer values representing the specified characters.~~
As correctly pointed out by [this comment](https://github.com/dotnet/csharplang/discussions/6350#discussioncomment-3346232), `Rune` would have to be used in that scenario.

The current way to compare against integer values is to cast the character into an integer, and then performing the pattern, like so
```csharp
return (int)c is >= 0x0250 and <= 0x02FF;
```
Which I find redundant, since the character can be losslessly upcast to an integer and reliably compared against literal integers that wouldn't overflow the character type.

The goal for me is to remove the need for that integer casting, meaning the following code would successfully compile into the same pattern matching expression:
```csharp
return c is >= 0x0250 and <= 0x02FF;
```

**Comment by huoyaoyuan** [comment]
In case you mentioned characters exceeding 16 bits, wouldn't the sample be based on `Rune` instead of `char`?

**Reply by Rekkonnect** [reply]
That's right, `Rune` would have to be used in those cases. I'll edit the original post to reflect the actual range of a char value.
